app/javascript/components/notifications/Widget.tsx

```diff
@@ -20,7 +20,18 @@ function reducer(state: WidgetProps, action: WidgetAction) {
 
 export function Widget({ count }: WidgetProps) {
   const [state, dispatch] = useReducer(reducer, { count: count })
-  const isUnread = state.count > 0
+  const variantClass = useMemo(() => {
+    switch (true) {
+      case state.count === 0:
+        return '--none'
+      case state.count >= 1 && state.count <= 9:
+        return '--digital'
+      case state.count >= 10 && state.count <= 99:
+        return '--double-digit'
+      case state.count >= 100:
+        return '--triple-digit'
+    }
```

Comment:
I've never seen this pattern before! Very interesting. What made you use this over regular if/else if statements?

Reply:
You can use Math too if you'd like:

```typescript
const CLASSES = ['--none', '--digital', '--double-digit', '--triple-digit']

...

const digits = Math.log(state.count) * Math.LOG10E + 1 | 0;
return CLASSES[digits]  || CLASSES[CLASSES.length - 1]
```

Reply:
Please add a comment if you do this!

Reply:
Hah yes!

Reply:
@ErikSchierboom I just like how every case is laid out nicely. But I don't have any preference if we go back to use if / else statements.